core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -231,6 +315,9 @@ package object config {
   private[spark] val BLOCK_MANAGER_PORT = ConfigBuilder("spark.blockManager.port")
     .doc("Port to use for the block manager when a more specific setting is not provided.")
     .intConf
+    .checkValue(v => v == 0 || (1024 <= v && v < 65536),
```

Comment:
Maybe we should let user to decide whether to bind to a privileged port. Taking an assumption here may potentially break some use cases.

Reply:
+1 on ^

Reply:
@jerryshao , @HyukjinKwon 
Because of the same port control at startServiceOnPort, do we need to open it here?

  def startServiceOnPort[T](
      startPort: Int,
      startService: Int => (T, Int),
      conf: SparkConf,
      serviceName: String = ""): (T, Int) = {

    require(startPort == 0 || (1024 <= startPort && startPort < 65536),
      "startPort should be between 1024 and 65535 (inclusive), or 0 for a random free port.")
     
     ...
}

Reply:
Ah, do you mean the check was already there and the current change just brings it ahead to make it fail fast? If so, that's fine.

Reply:
well, thanks!